addons/azureSets/azuresets.lua

```diff
@@ -122,22 +161,23 @@ function set_spells_from_spellset(spellset, setPhase)
     local slotToSetTo
     for i = 1, 20 do
         local slotName = 'slot%02u':format(i)
-        if currentSet[slotName] == nil then
+        if currentSpellSet[slotName] == nil then
             slotToSetTo = i
             break
         end
     end
 
     if slotToSetTo ~= nil then
         -- We found an empty slot. Find a spell to set.
-        for k,v in pairs(setToSet) do
-            if not currentSet:contains(v:lower()) then
+        for k, v in pairs(setToSet) do
+            if not currentSpellSet:contains(v:lower()) then
                 if v ~= nil then
                     local spellID = find_spell_id_by_name(v)
                     if spellID ~= nil then
```

Comment:
Same as before, you don't need the `spellID` check here, since it gets verified in `verify_and_set_spell` afterwards. In fact, if you take my previous suggestion about changing `find_spell_id_by_name` I would just integrate it into `verify_and_set_spell`, i.e. change `verify_and_set_spell` to take a string and slot, and search for the ID in there. That would also fit the description of the function and remove redundancy.